maltoolbox/model.py

```diff
@@ -864,4 +723,193 @@ def load_from_file(
             serialized_model = load_dict_from_json_file(filename)
         else:
             raise ValueError('Unknown file extension, expected json/yml/yaml')
-        return cls._from_dict(serialized_model, lang_classes_factory)
+        return cls._from_dict(serialized_model, lang_graph)
+
+
+class ModelAsset:
+    def __init__(self, name:str, lg_asset: LanguageGraphAsset):
+        self.name:str = name
+        self.id:Optional[int] = None
+        self.lg_asset:LanguageGraphAsset = lg_asset
+        self.defenses:dict = {}
+        self.extras:dict = {}
+        self.associations:list = []
+        self.attack_step_nodes:list = []
+        for step in self.lg_asset.attack_steps.values():
+            if step.type == 'defense':
+                self.defenses[step.name] = 1.0 if step.ttc and \
+                    step.ttc['name'] == 'Enabled' else 0.0
+
```

Comment:
So, we are moving away from the one-class-per-asset approach? Are all model assets going to be of type ModelAsset? Is this preferred?

Reply:
Yes, having them be different classes doesn't provide us with anything and it just makes certain checks harder.